src/main/java/task1/PhoneBook.java

```diff
@@ -0,0 +1,44 @@
+package task1;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PhoneBook {
+    final Map<String, Person> phoneBookEntries;
+
+    public PhoneBook() {
+        this.phoneBookEntries = new HashMap<>();
+    }
+
+    public void addPerson(String fullName, String phoneNumber, String address) {
+        if (fullName.isEmpty() || phoneNumber.isEmpty() || address.isEmpty()) {
+            throw new IllegalArgumentException("Full name, phone number, and address are mandatory fields.");
+        }
+        Person person = new Person(fullName, phoneNumber, address);
+        phoneBookEntries.put(fullName, person);
+    }
+
+    public Person findPerson(String fullName) throws EntryNotFoundException {
+        if (!phoneBookEntries.containsKey(fullName)) {
+            throw new EntryNotFoundException("Entry not found for: " + fullName);
+        }
+        return phoneBookEntries.get(fullName);
+    }
+
+    public void loadFromFile(String filename) {
```

Comment:
brakuje mi jeszcze metody do zapisywania do pliku. No bo teraz jest tak, że możesz sobie dodać nową osobę ale ona i tak przepadnie. Jak robisz odczytywanie z pliku w takim programie to rób też zapis, żeby aktualizować stan.